lms/djangoapps/certificates/management/commands/ungenerated_certs.py

```diff
@@ -112,10 +126,63 @@ def handle(self, *args, **options):
                         count, total, hours, minutes)
                     start = datetime.datetime.now(UTC)
 
-                if certificate_status_for_student(
-                        student, course_key)['status'] in valid_statuses:
+                cert_status = certificate_status_for_student(student, course_key)['status']
+                logger.info(
+                    (
+                        u"Student {user_id} has certificate status '{status}' "
+                        u"in course '{course}'"
+                    ).format(
+                        user_id=student.id,
+                        status=cert_status,
+                        course=unicode(course_key)
+                    )
+                )
+
+                if cert_status in valid_statuses:
+
                     if not options['noop']:
                         # Add the certificate request to the queue
                         ret = xq.add_cert(student, course_key, course=course)
```

Comment:
So it turns out add_cert does not do what you think it might.

It is actually a pretty complex function and logging in there might be useful as well.  This is the function where the actual call for grading happens: https://github.com/edx/edx-platform/blob/will/better-logging-for-cert-cmds/lms/djangoapps/certificates/queue.py#L206